src/main/java/com/code_intelligence/jazzer/mutation/mutator/collection/ChunkCrossOvers.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2023 Code Intelligence GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.code_intelligence.jazzer.mutation.mutator.collection;
+
+import com.code_intelligence.jazzer.mutation.api.PseudoRandom;
+import com.code_intelligence.jazzer.mutation.api.SerializingMutator;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+final class ChunkCrossOvers {
+  private ChunkCrossOvers() {}
+
+  static <T> void insertChunk(List<T> list, List<T> otherList, int maxSize, PseudoRandom prng) {
+    int maxChunkSize = Math.min(maxSize - list.size(), otherList.size());
+    int chunkSize = prng.closedRangeBiasedTowardsSmall(1, maxChunkSize);
+    int fromPos = prng.indexIn(otherList.size() - chunkSize);
+    int toPos = prng.indexIn(list.size());
+    List<T> chunk = otherList.subList(fromPos, fromPos + chunkSize);
+    list.addAll(toPos, chunk);
+  }
+
+  static <T> void overwriteChunk(List<T> list, List<T> otherList, PseudoRandom prng) {
+    int maxChunkSize = Math.min(list.size(), otherList.size());
+    int chunkSize = prng.closedRangeBiasedTowardsSmall(1, maxChunkSize);
+    int fromPos = prng.indexIn(otherList.size() - chunkSize);
+    int toPos = prng.indexIn(list.size() - chunkSize);
+    List<T> chunk = otherList.subList(fromPos, fromPos + chunkSize);
+    for (int i = 0; i < chunkSize; i++) {
+      list.set(toPos + i, chunk.get(i));
+    }
+  }
+
+  static <T> void crossOverChunk(
+      List<T> list, List<T> otherList, SerializingMutator<T> elementMutator, PseudoRandom prng) {
+    int maxChunkSize = Math.min(list.size(), otherList.size());
+    int chunkSize = prng.closedRangeBiasedTowardsSmall(1, maxChunkSize);
+    int fromPos = prng.indexIn(otherList.size() - chunkSize);
+    int toPos = prng.indexIn(list.size() - chunkSize);
+    List<T> chunk = otherList.subList(fromPos, fromPos + chunkSize);
+    for (int i = 0; i < chunkSize; i++) {
+      T value = list.get(toPos + i);
+      T otherValue = chunk.get(i);
+      list.set(toPos + i, elementMutator.crossOver(value, otherValue, prng));
+    }
+  }
+
```

Comment:
Yes, thought about that one as well but didn't add it in the end, as it's basically multiple `insertChunk` list cross overs. 

As it may very well yield good results, I suggest to wait with the implementation until we have somewhat meaningful benchmarks. WDYT?

Reply:
ah yeah we can do that :)